pyanaconda/network.py

```diff
@@ -277,6 +277,78 @@ def unset(self, *args):
         ifcfglog.debug("IfcfgFile.unset %s: %s", self.filename, args)
         SimpleConfigFile.unset(self, *args)
 
+def ensure_single_initramfs_connections():
+    """Ensure device configured in initramfs has no more than one NM connection.
+
+    In case of multiple connections for device having ifcfg configuration from
+    boot options, the connection should correspond to the ifcfg file.
+    NetworkManager can be generating additional in-memory connection in case it
+    fails to match device configuration to the ifcfg (#1433891).  By
+    reactivating the device with ifcfg connection the generated in-memory
+    connection will be deleted by NM.
+
+    Don't enforce on slave devices for which having multiple connections can be
+    valid (slave connection, regular device connection).
+    """
+
+    rv = []
+
+    for dev_name in nm.nm_devices():
+        try:
+            nm.nm_device_setting_value(dev_name, "connection", "uuid")
+        except nm.SettingsNotFoundError:
+            pass
+        except nm.MultipleSettingsFoundError:
+            if nm.nm_device_is_slave(dev_name):
+                continue
+
+            ifcfg_path = find_ifcfg_file_of_device(dev_name)
+            if not ifcfg_path:
+                log.error("network: multiple settings but no ifcfg for %s", dev_name)
+                continue
+
+            # Handle only ifcfgs created from boot options in initramfs
+            # (Kickstart based ifcfgs are handled in apply_kickstart)
+            if ifcfg_is_from_kickstart(ifcfg_path):
+                continue
+
+            ensure_active_ifcfg_connection_for_device(ifcfg_path, dev_name, only_replace=True)
+            rv.append(dev_name)
+
+    return rv
+
+def ensure_active_ifcfg_connection_for_device(ifcfg_path, dev_name, only_replace=False):
+    """Make sure active connection of a device is the one of ifcfg file
+
+    :param ifcfg_path: path of ifcfg file with the connection to be used
+    :type ifcfg_path: str
+    :param dev_name: name of device to apply the connection to
+    :type dev_name: str
+    :param only_replace: apply the connection only if the device has different
+                         active connection
+    :type only_replace: bool
+
+    """
+    msg = "not activating"
+    ifcfg = IfcfgFile(ifcfg_path)
+    ifcfg.read()
+    con_uuid = ifcfg.get("UUID")
+    active_con_uuid = nm.nm_device_active_con_uuid(dev_name)
+    if active_con_uuid or not only_replace:
```

Comment:
Hm, I think it does exactly what the documentation says.
If only_replace is True it will execute the if clause only if there is already an active connection. And in the if clause it will apply the connection it if it is different.